jwplatform/client.py

```diff
@@ -96,12 +97,48 @@ def request(self, method, path, body=None, headers=None, query_params=None):
 
 class _ScopedClient:
 
-    def __init__(self, client):
+    def __init__(self, client: JWPlatformClient):
         self._client = client
 
 
-class _ResourceClient(_ScopedClient):
+class _GenericResourceClient(_ScopedClient):
```

Comment:
This isn't more 'generic' than ResourceClient -- it adds a layer for sub-resources making it _less_ generic. _SubResourceClient might be more appropriate.

Reply:
Given that this only has a single use-case for one method though I'm not entirely sure it's necessary.